search-app/src/cmr/search/services/community_usage_metrics/metrics_service.clj

```diff
@@ -30,15 +33,70 @@
+                           :all-revisions? true
+                           :result-format :query-specified
+                           :result-fields [:short-name]})
+          results (qe/execute-query context query)
+          short-name (first (map :short-name (:items results)))]
+      short-name)))
+
+(defn- get-collection-by-short-name
+  "Query elastic for collection with a given short-name, returns short-name"
+  [context short-name]
+  (when (seq short-name)
+    (let [condition (qm/string-condition :short-name short-name)
+          query (qm/query {:concept-type :collection
+                           :condition condition
+                           :page-size 1
+                           :all-revisions? true
+                           :result-format :query-specified
+                           :result-fields [:short-name]})
+          results (qe/execute-query context query)
+          short-name (first (map :short-name (:items results)))]
+      short-name)))
+
+(defn- comprehensive-collection-short-name-search
+  "This function checks CMR for a collection with given short-name, if it can't find it
+   Then it will check for a collection with that entry-title and return its short-name.
+   Currently there are providers who are supply non-short-name values in short-name, which breaks usage
+   metrics.  This is a short term solution to this problem, the long term solution is correcting the
+   EMS data which may take a long time to happen."
+  [context short-name]
+  (if (get-collection-by-short-name context short-name)
+    short-name
+    (get-short-name-by-entry-title context short-name)))
+
+(defn- efficient-collection-short-name-search
+  "This function will check the current community-usage-metrics humanizer to see if this shortname exists,
+   if it does, then it will return that short-name, otherwise it will try looking up the short-name via
+   entry-title."
+  [context short-name current-metrics]
+  (if (get current-metrics short-name)
+    short-name
+    (comprehensive-collection-short-name-search context short-name)))
+
 (defn- csv-entry->community-usage-metric
   "Convert a line in the csv file to a community usage metric. Only storing short-name (product),
    version (can be 'N/A' or a version number), and access-count (hosts)"
-  [csv-line product-col version-col hosts-col]
+  [context csv-line product-col version-col hosts-col comprehensive current-metrics]
   (when (seq (remove empty? csv-line)) ; Don't process empty lines
    (let [short-name (read-csv-column csv-line product-col)
-         version (read-csv-column csv-line version-col)]
+         version (read-csv-column csv-line version-col)
+         searched-short-name (if (= "true" comprehensive)
+                              (comprehensive-collection-short-name-search context short-name)
+                              (efficient-collection-short-name-search context short-name current-metrics))]
      {:short-name (if (seq short-name)
```

Comment:
Move this check up to right after line 91 as a `when-not`, and move `version` and `searched-short-name` to its own `let` block.